### DISCUSSION TITLE
Option to hide out of stock products?

### DISCUSSION THREAD

**Discussion by rebelchris** [vendurehq/vendure#1679] [Q&A] [answered]
Hey everyone!

I wanted to know if it's possible to hide all out of stock products from the store front?

**Comment by michaelbromley** [comment, marked as answer]
Hi Chris!

Good question! Probably the best way to do this right now would be to create a plugin which uses the [EventBus](https://www.vendure.io/docs/typescript-api/events/event-bus/) to subscribe to the [StockMovementEvent](https://www.vendure.io/docs/typescript-api/events/event-types/#stockmovementevent). You can then define a function which looks up the current stockOnHand of the associated ProductVariant, and if it is `0`, it can set the `ProductVariant.enabled` to `false` using [ProductVariantService.update()](https://www.vendure.io/docs/typescript-api/services/product-variant-service/#update)

**Reply by rebelchris** [reply]
Thanks @michaelbromley that seems to work indeed 🙏
at first I thought it could simply be a on/off flag in a query.

Like only return result that have stock > 0
So on the other hand would it be a risk that people can order stock that's not actually in stock?

**Reply by michaelbromley** [reply]
There's no built-in support for a simple "on/off" toggle on this right now, mainly because it's just not been requested. I think in most cases, merchants want to keep showing a product even when out of stock, so that customers at least know about it and that they can buy it in future when stock returns.

> So on the other hand would it be a risk that people can order stock that's not actually in stock?

By default, if a customer attempts to `addItemToOrder` when there is insufficient stock, it will fail with a [InsufficientStockError](https://www.vendure.io/docs/graphql-api/shop/object-types/#insufficientstockerror). You can also have fine-grained control over this though, by configuring [back order support](https://www.vendure.io/docs/developer-guide/stock-control/#back-orders) per ProductVariant or globally.

Lastly, another potential option here is to use the [ElasticsearchPlugin](https://www.vendure.io/docs/typescript-api/elasticsearch-plugin/), which then allows you to [customize the search query logic](https://www.vendure.io/docs/typescript-api/elasticsearch-plugin/elasticsearch-options/#mapquery) so that it omits out-of-stock results from the returned result set. In this way you can prevent the product showing up in the search & collection list pages, whilst also keeping the actual product page available (not 404 like it would if disabled). This might be the optimal solution actually.
